### ISSUE TITLE
Features

### ISSUE THREAD

**Issue by h4nek** [h4nek/flink-rc#6]
The desired features of the implementation:

- [ ] Flink Timestamps
  - How should we use the timestamps, key by them w/ tolerance? What to do if we don't have them (have some default creation logic?)
  - Probably not part of the library but useful mostly for the examples.
- [ ] Flink State
  - Operator State (_CheckpointedState_) vs Keyed State (requires keyed streams - what should be the key?)

- [ ] Flink Checkpointing
  - Gives an ability to recover in case of system failure (fault tolerance).

**Non-mandatory (extra)**
- [ ] Parallelism (>1)
   - We should need keyed streams to make it work.

- [ ] Scalability - running on remote clusters?

- [ ] Rework Gradient Descent to be more efficient (streams?)

- [x] Gradient Descent Learning Rate Getting Smaller With Time (e.g. steps decay)
